helm-git-plugin.sh

```diff
@@ -237,47 +236,55 @@ main() {
   fi
 
   # Parse URI
-
   string_starts "$_raw_uri" "$url_prefix" ||
     error "Invalid format, got '$_raw_uri'. $error_invalid_prefix"
 
-  _raw_uri=$(echo "$_raw_uri" | sed 's/^git+//')
+  # Thanks to https://stackoverflow.com/a/63993578
+  readonly URI_REGEX='^([^:/?#]+):(//((([^/?#]+)@)?([^/?#]+)?))?(/([^?#]*))(\?([^#]*))?(#(.*))?$'
+  readonly _uri_scheme=$(echo "$_raw_uri" | sed -Ene "s'$URI_REGEX'\1'p")
+  readonly _uri_authority=$(echo "$_raw_uri" | sed -Ene "s'$URI_REGEX'\3'p")
+  readonly _uri_path=$(echo "$_raw_uri" | sed -Ene "s'$URI_REGEX'\8'p")
+  readonly _uri_query=$(echo "$_raw_uri" | sed -Ene "s'$URI_REGEX'\9'p")
 
-  git_proto=$(echo "$_raw_uri" | cut -d':' -f1)
-  readonly git_proto="$git_proto"
-  string_contains "$allowed_protocols" "$git_proto" ||
+  git_scheme=$(echo "$_uri_scheme" | sed -e 's/^git+//')
+  readonly git_scheme="$git_scheme"
+  string_contains "$allowed_protocols" "$git_scheme" ||
     error "$error_invalid_protocol"
 
-  git_repo=$(echo "$_raw_uri" | sed -E 's#^([^/]+//[^/]+[^@\?]+)@?[^@\?]+\??.*$#\1#')
-  readonly git_repo="$git_repo"
-  # TODO: Validate git_repo
+  git_repo_path=$(echo "${_uri_path}" | cut -d'@' -f 1)
+  readonly git_repo_path="$git_repo_path"
 
-  git_path=$(echo "$_raw_uri" | sed -E 's#.*@(([^\?]*)\/)?([^\?]*).*(\?.*)?#\1#' | sed -E 's/\/$//')
-  readonly git_path="$git_path"
-  # TODO: Validate git_path
+  git_file_path=$(echo "${_uri_path}" | cut -d'@' -f 2)
+  readonly git_file_path="$git_file_path"
 
-  helm_file=$(echo "$_raw_uri" | sed -E 's#.*@(([^\?]*)\/)?([^\?]*).*(\?.*)?#\3#')
+  helm_dir=$(dirname "${git_file_path}" | sed -r '/^[\.|/]$/d')
```

Comment:
```suggestion
  helm_dir=$(dirname "${git_file_path}" | sed -r '/^[\.|/]$/d' | sed 's/^\///')
```
optionally strip first /